src/Components/RelativeDateSelect.tsx

```diff
@@ -1,86 +1,76 @@
-  applyFn(text: string[]): void;
-  setTermAnchorEl(element: EventTarget | null): void;
-  setIntervalAnchorEl(element: EventTarget | null): void;
-}
-
-export const DateSelect: React.FC<Inputs> = (props) => {
-  enum menu {
-    term = 0,
-    interval = 1,
-  }
-
-  function getAnchorEl(identifier: number): Element {
-    if (identifier == menu.term) {
-      return props.termAnchorEl;
-    } else {
-      return props.intervalAnchorEl;
-    }
-  }
-
-  function setAnchorEl(identifier: number, item: EventTarget | null): void {
-    if (identifier == menu.term) {
-      props.setTermAnchorEl(item);
-    } else {
-      props.setIntervalAnchorEl(item);
-    }
-  }
-
-  function handleMenuClick(identifier: number, event: EventTarget): void {
-    setAnchorEl(identifier, event);
-  }
-
-  function handleClose(identifier: number, item: string): void {
-    setAnchorEl(identifier, null);
-    console.log(item)
-    let words = props.relativeSelectContent;
-    words[identifier] = item;
-    props.applyFn(words);
-  }
-
-  function getDropdownObj(index: number) {
-    return {
-      relativeTerms: props.relativeTerms,
-      relativeIntervals: props.relativeIntervals,
-      relativeSelectContent: props.relativeSelectContent,
-      handleClose,
-      classes: props.classes,
-      handleMenuClick,
-      getAnchorEl,
-      identifier: index,
-    };
-  }
-
-  return (
-    <Box className={props.classes.flexRow}>
-      <Box>
-        <RelativeDateSelectDropdown {...getDropdownObj(0)} />
-      </Box>
-      <Box ml={1}>
-        <RelativeDateSelectDropdown {...getDropdownObj(1)} />
-      </Box>
-    </Box>
-  );
-};
+import React, { useContext } from "react";
+import { DateRange, TermContext } from "./DateRange";
+import { Box } from "@material-ui/core";
+import { RelativeDateSelectDropdown } from "./RelativeDateSelectDropdown";
+import { DropdownData } from "./Types";
+import { GlobalContext } from "./Constants";
+
+interface Inputs {
+  classes: any;
+  dateRange: DateRange;
+  applyFn(text: number): void;
+  dropdownData: DropdownData;
+  setDropdownData(data: DropdownData): void;
+}
+
+export const DateSelect: React.FC<Inputs> = (props) => {
+  const globals = useContext(GlobalContext);
+
+  enum menu {
+    term = 0,
+    interval = 1,
+  }
+
+  function getAnchorEl(identifier: number): Element {
+    console.log(props.dropdownData.intervalAnchorEl);
+    if (identifier == menu.term) {
+      return props.dropdownData.termAnchorEl;
+    } else {
+      return props.dropdownData.intervalAnchorEl;
+    }
+  }
+
+  function setAnchorEl(identifier: number, item: EventTarget | null): void {
+    if (identifier == menu.term) {
+      props.setDropdownData({ ...props.dropdownData, termAnchorEl: item });
+    } else {
+      props.setDropdownData({ ...props.dropdownData, intervalAnchorEl: item });
+    }
+  }
+
+  function handleMenuClick(identifier: number, event: EventTarget): void {
+    console.log(props.dropdownData.intervalAnchorEl);
```

Comment:
Check the PR for any other lingering console.log statements.

Reply:
Done